hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -555,10 +564,10 @@ private String functionCost() {
     for (CostFunction c : costFunctions) {
       builder.append(c.getClass().getSimpleName());
       builder.append(" : (");
-      if (c.isNeeded()) {
-        builder.append(c.getMultiplier());
+      if (c.isNeeded() || c.getMultiplier() > 0) {
+          builder.append("multiplier=" + c.getMultiplier());
```

Comment:
This smells like an error in how `isNeeded` gets implemented. It should be taking into account things like "the multiplier isn't >0". I don't think this is an issue with your specific PR since it happens all over the `StochasticLoadBalancer` class. Something to clean up later maybe.

Reply:
Actually this is not needed. There are some confusion in other part of the code on negative multipliers but it shouldn't happen. Will clean up.